### PR TITLE
yarn rw check throws on failure

### PR DESCRIPTION
**Pull request by szainab**
Fix for #3906

This is my first contribution to a project. Please let me know if I need to fix anything/if I did it wrong.

### CODE REVIEW

File: packages/structure/src/index.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export async function printDiagnostics(
     }
   } catch (e: any) {
     console.log('runtime error: ' + e.message)
+    throw e
```
Comment (thedavidprice) [inline packages/structure/src/index.ts:40]:
```suggestion
    throw new Error(e)
```

**Comment by thedavidprice** [comment]
@szainab Welcome to the Redwood Community! We've very excited you're taking a step into open source contributions. Apologies for the delay here as we got lost a bit in the holiday shuffle.

Please see my comments on your code. And do let me know if you have any questions as you go. Happy to help!

**Comment by thedavidprice** [comment]
@szainab checking back here — did you have any questions about my comments?

**Comment by szainab** [comment]
Hi @thedavidprice , I seem to have missed your comments. Sorry about that. Your comment makes sense, I'll update the PR tonight :)

**Comment by thedavidprice** [comment]
(gentle) nudge @szainab 

Just want to see if there's anything blocking (i.e. questions or code not making sense). Grately appreciate any/all time you can put into this. And understand schedules can get busy.

**Comment by szainab** [comment]
Hi David, really sorry, I've updated my PC and having some trouble with the setup. When I run yarn, I'm getting the following error: 

```
➤ YN0001: │ Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
    at new NodeError (node:internal/errors:371:5)
    at PassThrough.onclose (node:internal/streams/end-of-stream:134:30)
    at PassThrough.emit (node:events:520:28)
    at emitCloseNT (node:internal/streams/destroy:145:10)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)```
    
   Trying to solve this.

**Comment by thedavidprice** [comment]
Hi @szainab Thank you for the continued work on this (and apologies for the delayed reply). The logic looks really good so I spent some time checking out the functionality. What I do for a CLI command like this is use this testing option (found in the CONTRIBUTING.md doc):
- [Testing the CLI in Your Project](https://github.com/redwoodjs/redwood/blob/main/CONTRIBUTING.md#testing-the-cli-in-your-project)

So after building the project, I `cd packages/cli` and run `yarn dev check --cwd <path/to/my/test/project>`. I removed a query from a Service (error) and a name from a Cell query (warning).

Most of the changes I made were aesthetic except for swapping a throw for a process.exit. Here's what it looked like with the throw:
Screen Shot 2022-02-15 at 10 09 48 PM

I thought the full error object and message were unhelpful here because 1) the command itself wasn't failing at this point and 2) the output already contained the error message and trace.

So here's what the output looks like across all the cases.

**Success**
Screen Shot 2022-02-15 at 10 10 55 PM

**Error and Warning**
Screen Shot 2022-02-15 at 10 10 13 PM

**Warning
Screen Shot 2022-02-15 at 10 10 32 PM

---

Great work and a huge help on this! I'm merging it after CI passes.

**Review by thedavidprice — APPROVED** [review]

**Pull request merged**